src/app/components/User/User.tsx

```diff
@@ -6,11 +7,18 @@ function openLoginDialog() {
   overwolf.profile.openLoginDialog();
 }
 
-const User = (): JSX.Element => {
+type UserProps = {
+  onUserClick: (username: Profile['username']) => void;
+};
+
+const User = ({ onUserClick }: UserProps): JSX.Element => {
   const { currentUser, errorMessage } = useCurrentUser();
 
   return (
-    <section className={classes.container}>
+    <section
+      className={classes.container}
+      onClick={() => (currentUser ? onUserClick(currentUser.username) : null)}
```

Comment:
You don't need the arrow function if there is no `currentUser`.
Does this work?
```suggestion
      onClick={currentUser  && () => onUserClick(currentUser.username))}
```
Otherwise:
```suggestion
      onClick={currentUser  ?  () => onUserClick(currentUser.username)) : undefined}
```

Reply:
It doesn't work cause it's saying: `Type 'null' is not assignable to type 'MouseEventHandler<HTMLElement> | undefined'.`